raiden/tests/unit/test_channelstate.py

```diff
@@ -1579,3 +1580,67 @@ def test_update_must_be_called_if_close_lost_race():
     )
     iteration = channel.handle_channel_closed(channel_state, state_change)
     assert must_contain_entry(iteration.events, ContractSendChannelUpdateTransfer, {})
+
+
+def test_update_transfer():
+    """ This tests that receiving an update transfer event for a
+    closed channel sets the update_transaction member
+    """
+    our_model1, _ = create_model(70)
+    partner_model1, privkey2 = create_model(100)
+    channel_state = create_channel_from_models(our_model1, partner_model1)
+
+    pseudo_random_generator = random.Random()
+    block_number = 10
+    state_change = ActionChannelClose(
+        channel_state.token_network_identifier,
+        channel_state.identifier,
+    )
+    iteration = channel.state_transition(
+        channel_state,
+        state_change,
+        pseudo_random_generator,
+        block_number,
+    )
+
+    update_transfer_state_change = ContractReceiveUpdateTransfer(
+        partner_model1.participant_address,
+        channel_state.token_network_identifier,
+        channel_state.identifier,
+        23,
+    )
+
+    # update_transaction in channel state should not be set
+    channel_state = iteration.new_state
+    assert channel_state.update_transaction is None
+
+    update_block_number = 13
+    iteration2 = channel.handle_channel_updated_transfer(
```

Comment:
this is weird, the channel is opened in the test, and update transfer may only be called [on a closed channel](https://github.com/raiden-network/raiden-contracts/blob/a2f9abbf0d0964b3e73fbd24dc09bcfbfaa94b1a/raiden_contracts/contracts/TokenNetwork.sol#L508).

Please fix the order of the state changes.